Packs/Netcraft_V2/Integrations/Netcraft/Netcraft.py

```diff
@@ -0,0 +1,1142 @@
+        return self.submission_http_request(
+            'GET', f'api/v3/submission/{submission_uuid}/mail',
+        )
+
+    def mail_screenshot_get(self, submission_uuid: str) -> requests.Response:
+        return self.submission_http_request(
+            'GET', f'api/v3/submission/{submission_uuid}/mail/screenshot',
+            resp_type='response', ok_codes=(200, 404)
+        )
+
+    def email_report_submit(self, body: dict) -> dict:
+        return self.submission_http_request(
+            'POST', 'api/v3/report/mail', json_data=body,
+        )
+
+    def submission_url_list(self, submission_uuid, params: dict) -> dict:
+        return self.submission_http_request(
+            'GET', f'api/v3/submission/{submission_uuid}/urls', params=params,
+        )
+
+    def url_screenshot_get(self, submission_uuid: str, url_uuid: str, screenshot_hash: str) -> requests.Response:
+        return self.submission_http_request(
+            'GET', f'api/v3/submission/{submission_uuid}/urls/{url_uuid}/screenshots/{screenshot_hash}',
+            resp_type='response',
+        )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def sub_dict(d: dict, *keys) -> dict:
+    '''Returns a sub dict of d with the given keys'''
+    return {k: d.get(k) for k in keys}
+
+
+def pop_keys(d: dict, *keys) -> dict:
+    '''Removes the given keys of d and returns them as their own dict'''
+    return {k: d.pop(k, None) for k in keys}
+
+
+def int_to_readable_bool(val: str | int | None) -> str | None:
+    '''
+    Converts the 1 or 0 values returned by netcraft to a human readable string.
+    The val provided can be an int, str or None (in which case None should be returned).
+    '''
+    return {'0': 'No', '1': 'Yes'}.get(str(val))
+
+
+def int_to_bool(val: str | int | None) -> Any:
+    '''
+    Converts the 1 or 0 values returned by netcraft to a boolean for the context, if possible.
+    The val provided can be an int, str or None (in which case None should be returned).
+    '''
+    return {'0': False, '1': True}.get(str(val), val)
+
+
+def convert_binary_keys_to_bool(d: dict, *keys):
+    '''Converts values of specified keys in a dictionary to booleans.
+
+    This takes a dictionary and list of keys, and converts the values of those keys
+    in the dictionary to True/False based on their string or int values of "0", "1", 0 or 1.
+
+    Args:
+        d (dict): The dictionary to modify in-place.
+        *keys: The list of keys in the dictionary to convert.
+
+    Returns:
+        None
+
+    Examples:
+        >>> d = {'a': '1', 'b': '0'}
+        >>> convert_binary_keys_to_bool(d, 'a', 'b')
+        >>> print(d)
+        {'a': True, 'b': False}
+    '''
+    d |= {
+        key: int_to_bool(d.get(key))
+        for key in keys
+    }
+
+
+def read_base64encoded_file(filepath: str) -> str:
+    '''Returns a base64 encoded string of the file'''
+    with open(filepath, 'rb') as f:
+        return base64.b64encode(f.read()).decode('ascii')
```

Comment:
You're right, ascii isn't needed. It works just fine without any arg.
the decoding just turns it into a string so that it can be sent to the API, it doesn't have any connection with the original file format.